### PR TITLE
APEX-1562 - send welcome message to birthday squad

### PR DESCRIPTION
**Pull request by BaDo2001**
Issue: https://apexlab.atlassian.net/browse/APEX-1562

- collects icebreaker threads and presentIdeas and sends it to the birthday squad conversation
- fixes issue with multiple pr-s deploying same rule for default event bus
- adds manual testing for createSquad

### CODE REVIEW

File: packages/core/services/slack/constructBirthdaySquadWelcomeMessage.ts [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+  presentIdeas
+    .map((idea) =>
+      idea
+        .split("\n")
+        .map((line) => `> ${line}`)
+        .join("\n"),
+    )
+    .join("\n\n");
+
+export const constructBirthdaySquadWelcomeMessage = async ({
```
Comment (debrecenid) [inline packages/core/services/slack/constructBirthdaySquadWelcomeMessage.ts]: This should be responsible only for createing the message body, just like the outher constuct, dont have to be do async opartions, i think the sendSquadWelcomeMessage is a better place for that, and  just pass the neccessary object to this function.

File: packages/core/types/schedule/index.ts [hunk 2]
```diff
@@ -1,3 +1 @@
-export const scheduleEvent = "scheduleEvent";
-
-export type ScheduleEventType = typeof scheduleEvent;
+export const getScheduledEvent = (stage: string) => `scheduleEvent-${stage}`;
```
Comment (debrecenid) [inline packages/core/types/schedule/index.ts]: This should be moved to the utils until its not a type anymore.

File: packages/core/types/cron/index.ts [hunk 3]
```diff
@@ -1,4 +1,4 @@
-export const iceBreaker = "iceBreaker";
-export const daily = "daily";
+export type CronEventType = "iceBreaker" | "daily";
 
-export type CronEventType = typeof iceBreaker | typeof daily;
+export const getCronEvent = (type: CronEventType, stage: string) =>
```
Comment (debrecenid) [inline packages/core/types/cron/index.ts]: Same as getScheduledEvent,this should be moved to the utils until its not a type anymore, but atleast this one have some relation to a type.

File: tests/integration/askPresentAndSquadJoin.test.ts [hunk 4]
```diff
@@ -62,9 +61,9 @@ describe("Present and Squad Join", () => {
         },
       ]);
 
-      const eventId = "PI1_" + Date.now().toString();
+      const eventId = "SJ1_" + Date.now().toString();
```
Comment (debrecenid) [inline tests/integration/askPresentAndSquadJoin.test.ts:64]: Thank you! 😅

File: tests/integration/welcomeMessage.test.ts [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+vi.mock("@/services/slack/createSlackApp", async () => ({
+  createSlackApp: vi.fn().mockReturnValue(
+    new App({
+      signingSecret: import.meta.env.VITE_SLACK_SIGNING_SECRET,
+      token: import.meta.env.VITE_SLACK_BOT_TOKEN,
+    }),
+  ),
+}));
+
+describe("Create birthday squad", () => {
```
Comment (debrecenid) [inline tests/integration/welcomeMessage.test.ts]: This should be Send welcome message, also the file name should be sendSquadWelcomeMessage after the lambda, just like most of the other tests.

File: packages/core/services/slack/constructBirthdaySquadWelcomeMessage.ts [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+      },
+      {
+        type: "section",
+        text: {
+          type: "mrkdwn",
+          text: `*Icebreaker threads*\n${iceBreakerLinks}`,
+        },
+      },
+      {
+        type: "section",
```
Comment (debrecenid) [inline packages/core/services/slack/constructBirthdaySquadWelcomeMessage.ts:64]: Should we skip this section if there is no presentIdea?

**Review by debrecenid — CHANGES_REQUESTED** [review]
Left some comment, and also unit tests are missing. The integration test are so complex, but i gues some unit test would be nice, the catch the erorrs early on, and for the other functions we also have both kind of tests.

File: packages/core/services/slack/constructBirthdaySquadWelcomeMessage.ts (hunk 1 above)
Comment (BaDo2001) [reply packages/core/services/slack/constructBirthdaySquadWelcomeMessage.ts]: Ur right, this was done before I rebased on your branch.

File: packages/core/types/schedule/index.ts (hunk 2 above)
Comment (BaDo2001) [reply packages/core/types/schedule/index.ts]: Ur right, moved it and also renamed it to getScheduleDetailType

File: packages/core/types/cron/index.ts (hunk 3 above)
Comment (BaDo2001) [reply packages/core/types/cron/index.ts]: Ur right, moved it and also renamed it to getCronDetailType

File: tests/integration/welcomeMessage.test.ts (hunk 5 above)
Comment (BaDo2001) [reply tests/integration/welcomeMessage.test.ts]: Renamed it, also I used to publish the createBirthdaySquad but it's better to publish the sendSquadWelcomeMessage directly.

File: packages/core/services/slack/constructBirthdaySquadWelcomeMessage.ts (hunk 6 above)
Comment (BaDo2001) [reply packages/core/services/slack/constructBirthdaySquadWelcomeMessage.ts:64]: It shouldn't happen tbh. 😄 But I changed it.

**Review by debrecenid — APPROVED** [review]

**Pull request merged**